apiserver/facades/client/applicationoffers/base.go

```diff
@@ -53,31 +54,24 @@ func (api *BaseAPI) checkAdmin(
 
 // checkControllerAdmin ensures that the logged in user is a controller admin.
 func (api *BaseAPI) checkControllerAdmin(ctx context.Context) error {
-	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, api.ControllerModel.ControllerTag())
+	return api.Authorizer.HasPermission(ctx, permission.SuperuserAccess, api.controllerTag)
 }
 
 // modelForName looks up the model details for the named model and returns
 // the model (if found), the absolute model path which was used in the lookup,
 // and a bool indicating if the model was found,
-func (api *BaseAPI) modelForName(modelName, ownerName string) (Model, string, bool, error) {
+func (api *BaseAPI) modelForName(ctx context.Context, modelName, ownerName string) (*model.Model, string, error) {
 	modelPath := fmt.Sprintf("%s/%s", ownerName, modelName)
-	var model Model
-	uuids, err := api.ControllerModel.AllModelUUIDs()
+	models, err := api.modelService.ListAllModels(ctx)
 	if err != nil {
-		return nil, modelPath, false, errors.Trace(err)
+		return nil, modelPath, errors.Trace(err)
 	}
-	for _, uuid := range uuids {
-		m, release, err := api.StatePool.GetModel(uuid)
-		if err != nil {
-			return nil, modelPath, false, errors.Trace(err)
-		}
-		defer release()
-		if m.Name() == modelName && m.Owner().Id() == ownerName {
-			model = m
-			break
+	for _, m := range models {
+		if m.Name == modelName && m.OwnerName.Name() == ownerName {
+			return &m, modelPath, nil
```

Comment:
Do we have a get service method instead of list everything?

Reply:
I wasn't able to find any that gets a model based on name and owner name. I was thinking about implementing it in state, but based on Tom's initial review in Lye's PR,  modifying state doesn't seem to be in the scope of this task.

Reply:
Please implement the service. It makes sense and we already have the state layer implemented.

Reply:
Yup sure, will implement as discussed. Thanks Kelvin and Tom :D